### DISCUSSION TITLE
database modification after system migration

### DISCUSSION THREAD

**Discussion by giostark** [Taiko2k/Tauon#1326] [Q&A]
Hi!
After some issue (resolved) in the past I can say this is the best player EVER used. I love it.
So I still try to bring it with my desktop at any change of software or hardware.
Now , I changed M2 drive. Reinstalled the entire system.  
I copied the old local data folder of Tauon in my new home. I preserved all the paths of  the drives (even in the fstab)
Everything showed up perfectly...but ! The name of the user is different :-(
If I right click on the song and try to open the folder it show (correctly) the old path with the old user.
All the path are the same except for user name.
I have seen that the database is state.p and state.p.backup and are located in local home folder.
I tried to open the file with kate (editor) and tried to substitute /home/server/ (old user) with /home/ws/ (new user) and save the file , but it came altered and didnt work.
I tried also DBgate and PyCharm but the file.p still have a format not readable.  
There is a way for exchange those names ? playlist and library were not exactly small :-) 
Thanks for your time !

**Comment by Taiko2k** [comment]
Its possible to edit it manually, but it has to be done programmatically. Use Python `a = pickle.load(path)`, edit it then `pickle.dump(a, path)` to save.

The track db is `a[162]`, which is a list of dictionaries. Keys "fullpath" and "parent_folder_path" are ones that come to mind you'd want to do a find and replace on. 

Maybe you could to a convert to JSON -> find and replace the whole thing, then JSON back to Python back to Pickle.

**Comment by RSKYS** [comment]
There's even an easier fix for this; if you encounter with this in future, just make a symlink through /home

```
sudo rm -rf /home/server; \
( cd /home/ws; sudo chown -R ws:users . ); \
sudo ln -sf /home/ws /home/server  # You could ignore two above if there was no issue before and after
```
It fools any program around which looking for same old paths, easy pisy.
